edge/yaml/yaml.go

```diff
@@ -34,7 +35,11 @@ func NewYAML(fileContent string, credentials []agent.RegistryCredentials) *yaml
 func (y *yaml) getRegistryCredentialsByImageURL(imageURL string) []agent.RegistryCredentials {
 	credentials := []agent.RegistryCredentials{}
 	for _, r := range y.registryCredentials {
-		if strings.Contains(imageURL, r.ServerURL) {
+		domain, _, err := getRegistryAndPath(imageURL)
+		if err != nil {
+			return nil
+		}
+		if strings.Contains(domain, r.ServerURL) || strings.Contains(r.ServerURL, domain) {
```

Comment:
Might be good to explain this logic with a comment.   It basically sanitises the registry imageUrl.

e.g. 
imageUrl = prabhat/nginx   when passed through getRegistryAndPath will return "docker.io" for the domain.
While  imageUrl=quay.io/prabhat/nginx will return "quay.io" for the domain

Second check, we really just need to ensure the domain is in the serverUrl.